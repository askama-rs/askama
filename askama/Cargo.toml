[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
