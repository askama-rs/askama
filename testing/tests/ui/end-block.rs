use askama::Template;

#[derive(Template)]
#[template(
    ext = "html",
    source = "{% if x %}{% if x %}{% endif %}",
)]
struct EndIf {
    x: bool,
}

#[derive(Template)]
#[template(
    ext = "html",
    source = "{% if x %}",
)]
struct EndIf2 {
    x: bool,
}

#[derive(Template)]
#[template(
    ext = "html",
    source = "{% match x %}",
)]
struct EndMatch {
    x: bool,
}

#[derive(Template)]
#[template(
    ext = "html",
    source = "{% for a in x %}",
)]
struct EndFor {
    x: [u32; 2],
}

#[derive(Template)]
#[template(
    ext = "html",
    source = "{% macro bla %}",
)]
struct EndMacro;

#[derive(Template)]
#[template(
    ext = "html",
    source = "{% filter bla %}",
)]
struct EndFilter;

#[derive(Template)]
#[template(
    ext = "html",
    source = "{% block bla %}",
)]
struct EndBlock;

fn main() {
}
