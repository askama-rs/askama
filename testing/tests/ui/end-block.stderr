error: expected `endif` to terminate `if` node, found nothing
 --> <source attribute>:1:3
       "if x %}{% if x %}{% endif %}"
 --> tests/ui/end-block.rs:6:14
  |
6 |     source = "{% if x %}{% if x %}{% endif %}",
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `endif` to terminate `if` node, found nothing
 --> <source attribute>:1:3
       "if x %}"
  --> tests/ui/end-block.rs:15:14
   |
15 |     source = "{% if x %}",
   |              ^^^^^^^^^^^^

error: expected `endmatch` to terminate `match` node, found nothing
 --> <source attribute>:1:3
       "match x %}"
  --> tests/ui/end-block.rs:24:14
   |
24 |     source = "{% match x %}",
   |              ^^^^^^^^^^^^^^^

error: expected `endfor` to terminate `for` node, found nothing
 --> <source attribute>:1:3
       "for a in x %}"
  --> tests/ui/end-block.rs:33:14
   |
33 |     source = "{% for a in x %}",
   |              ^^^^^^^^^^^^^^^^^^

error: expected `endmacro` to terminate `macro` node, found nothing
 --> <source attribute>:1:3
       "macro bla %}"
  --> tests/ui/end-block.rs:42:14
   |
42 |     source = "{% macro bla %}",
   |              ^^^^^^^^^^^^^^^^^

error: expected `endfilter` to terminate `filter` node, found nothing
 --> <source attribute>:1:3
       "filter bla %}"
  --> tests/ui/end-block.rs:49:14
   |
49 |     source = "{% filter bla %}",
   |              ^^^^^^^^^^^^^^^^^^

error: expected `endblock` to terminate `block` node, found nothing
 --> <source attribute>:1:3
       "block bla %}"
  --> tests/ui/end-block.rs:56:14
   |
56 |     source = "{% block bla %}",
   |              ^^^^^^^^^^^^^^^^^
